### PR TITLE
Handle possible errors in `core-processor`

### LINKED ISSUE #761: Check usage of `Result/Option` in `process_queue`
### File Location(s)

_No response_

### Proposal

Process all errors in idiomatic Rust-way

### LINKED ISSUE #768: Check usage of `Result/Option` in `core-processor`
### File Location(s)

_No response_

### Proposal

Move checks on type level or describe why panic cannot happen

### PR DESCRIPTION
**Pull request by gshep**
Resolves #768 

- eliminate `expect` in `process`
- remove `unwrap` in `EnvExt::alloc`
- remove `unreachable` in `process_success`
- move `process_many` to `tests/common` (it's not called by other modules)

@gear-tech/dev

### CODE REVIEW

File: core-processor/src/processor.rs [hunk 1]
```diff
@@ -25,116 +25,49 @@ use crate::{
...
-use alloc::{collections::BTreeMap, vec::Vec};
+use alloc::vec::Vec;
 use gear_backend_common::{Environment, IntoExtInfo};
 use gear_core::{
     env::Ext as EnvExt,
     message::{Dispatch, DispatchKind, ExitCode, Message},
     program::ProgramId,
 };
 
+enum SuccessfulDispatchResultKind {
```
Comment (breathx) [inline core-processor/src/processor.rs:36]: providing from/into from common dispatch result kind allows removing strange matching before `process_success` call

**Review by breathx — APPROVED** [review]

File: core-processor/src/processor.rs (hunk 1 above)
Comment (shamilsan) [reply core-processor/src/processor.rs:36]: > providing from/into from common dispatch result kind allows removing strange matching before `process_success` call

As I see this `From` implementation will require empty branch (`_ => ...`) processing. This yields either silencing the wrong enum option or return the `unreachable()` back.

File: core-processor/src/processor.rs [hunk 2]
```diff
@@ -198,17 +131,31 @@ fn process_error(
 }
 
 /// Helper function for journal creation in success case
-fn process_success(res: DispatchResult) -> Vec<JournalNote> {
+fn process_success(
+    kind: SuccessfulDispatchResultKind,
+    DispatchResult {
```
Comment (shamilsan) [inline core-processor/src/processor.rs:136]: I'd move the destructuring inside the function.

**Review by shamilsan — APPROVED** [review]
Great job 👍

File: core-processor/src/processor.rs (hunk 1 above)
Comment (gshep) [reply core-processor/src/processor.rs:36]: This is intentional since other methods require `unreachable`/`panic`/etc

**Pull request merged**